Formula/vasm.rb

```diff
@@ -0,0 +1,60 @@
+class Vasm < Formula
+  desc "Portable and retargetable assembler"
+  homepage "http://sun.hasenbraten.de/vasm/"
+  url "http://server.owl.de/~frank/tags/vasm1_7e.tar.gz"
+  version "1.7e"
+  sha256 "2878c9c62bd7b33379111a66649f6de7f9267568946c097ffb7c08f0acd0df92"
+
+  # target options
+  option "with-6502", "Enable 6502 CPU target (binary vasm6502_SYNTAX)"
+  option "with-6800", "Enable Motorola 6800 CPU target (binary vasm6800_SYNTAX)"
+  option "with-arm", "Enable ARM CPU target (binary vasmarm_SYNTAX)"
+  option "with-c16x", "Enable c16x/st10 microcontroller target (binary vasmc16x_SYNTAX)"
+  option "with-jagrisc", "Enable Atari Jaguar GPU/DSP RISC target (binary vasmjagrisc_SYNTAX)"
+  option "with-m68k", "Enable Motorola 68K CPU target (binary vasmm68k_SYNTAX)"
+  option "with-ppc", "Enable PowerPC CPU target (binary vasmppc_SYNTAX)"
+  option "with-tr3200", "Enable Trillek TR3200 CPU target (binary vasmtr3200_SYNTAX)"
+  option "with-vidcore", "Enable VideoCore IV CPU target (binary vasmvidcore_SYNTAX)"
+  option "with-x86", "Enable Intel x86 CPU target (binary vasmx86_SYNTAX)"
+  option "with-z80", "Enable Zilog Z80 CPU target (binary vasmz80_SYNTAX)"
+
+  # syntax options
+  option "with-std", "Enable standard syntax (binary vasmCPU_std)"
+  option "with-madmac", "Enable MadMac (Atari) syntax (binary vasmCPU_madmac)"
+  option "with-mot", "Enable Motorola syntax (binary vasmCPU_mot)"
+  option "with-oldstyle", "Enable oldstyle (8-bit) syntax (binary vasmCPU_oldstyle)"
+
+  def install
+    configs = 0
+    %w[6502 6800 arm c16x jagrisc m68k ppc tr3200 vidcore x86 z80].select { |c| build.with? c }.each do |cpu|
+      %w[std madmac mot oldstyle].select { |s| build.with? s }.each do |syntax|
+        prog = "vasm#{cpu}_#{syntax}"
+        system "make", prog, "CPU="+cpu, "SYNTAX="+syntax
+        bin.install prog
+        configs += 1
+      end
+    end
+
+    odie "Please specify at least one cpu with --with-<cpu> and at least one syntax with --with-<syntax>" unless configs > 0
```

Comment:
Because of the nature of the project (a cross-assembler), it is not really possible to pick defaults. Each choice of CPU is a distinct assembler.

An alternative to options would be to produce N x M formulae, which doesn't seem an improvement.